pvlib/tests/iotools/test_psm3.py

```diff
@@ -50,47 +65,47 @@ def assert_psm3_equal(header, data, expected):
 
 
 @needs_pandas_0_22
-def test_get_psm3_tmy():
+@pytest.mark.flaky(reruns=5, reruns_delay=2)
+def test_get_psm3_tmy(nrel_api_key):
     """test get_psm3 with a TMY"""
-    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL,
-                                 names='tmy-2017')
+    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, nrel_api_key,
+                                 PVLIB_EMAIL, names='tmy-2017')
     expected = pd.read_csv(TMY_TEST_DATA)
     assert_psm3_equal(header, data, expected)
-    # check errors
-    with pytest.raises(HTTPError):
-        # HTTP 403 forbidden because api_key is rejected
-        psm3.get_psm3(LATITUDE, LONGITUDE, api_key='BAD', email=PVLIB_EMAIL)
-    with pytest.raises(HTTPError):
-        # coordinates were not found in the NSRDB
-        psm3.get_psm3(51, -5, DEMO_KEY, PVLIB_EMAIL)
-    with pytest.raises(HTTPError):
-        # names is not one of the available options
-        psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL, names='bad')
 
 
 @needs_pandas_0_22
-def test_get_psm3_singleyear():
+@pytest.mark.flaky(reruns=5, reruns_delay=2)
+def test_get_psm3_singleyear(nrel_api_key):
     """test get_psm3 with a single year"""
-    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL,
-                                 names='2017', interval=30)
+    header, data = psm3.get_psm3(LATITUDE, LONGITUDE, nrel_api_key,
+                                 PVLIB_EMAIL, names='2017', interval=30)
     expected = pd.read_csv(YEAR_TEST_DATA)
     assert_psm3_equal(header, data, expected)
-    # check leap day
-    _, data_2012 = psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL,
-                                 names='2012', interval=60, leap_day=True)
-    assert len(data_2012) == (8760+24)
-    # check errors
-    with pytest.raises(HTTPError):
-        # HTTP 403 forbidden because api_key is rejected
-        psm3.get_psm3(LATITUDE, LONGITUDE, api_key='BAD', email=PVLIB_EMAIL,
-                      names='2017')
-    with pytest.raises(HTTPError):
-        # coordinates were not found in the NSRDB
-        psm3.get_psm3(51, -5, DEMO_KEY, PVLIB_EMAIL, names='2017')
-    with pytest.raises(HTTPError):
-        # intervals can only be 30 or 60 minutes
-        psm3.get_psm3(LATITUDE, LONGITUDE, DEMO_KEY, PVLIB_EMAIL, names='2017',
-                      interval=15)
+
+
+@pytest.mark.parametrize('latitude, longitude, api_key, names, interval',
+                         [(LATITUDE, LONGITUDE, 'BAD', 'tmy-2017', 60),
+                          (51, -5, nrel_api_key, 'tmy-2017', 60),
+                          (LATITUDE, LONGITUDE, nrel_api_key, 'bad', 60),
+                          (LATITUDE, LONGITUDE, nrel_api_key, '2017', 15),
+                          ])
+@needs_pandas_0_22
+@pytest.mark.flaky(reruns=5, reruns_delay=2)
+def test_get_psm3_tmy_errors(
+    latitude, longitude, api_key, names, interval
+):
+    """Test get_psm3() for multiple error scenarios:
```

Comment:
NIT: I'm not a fan of this docstring style, and it seems inconsistent with pvlib, [pep8](https://pep8.org/#documentation-strings), and other packages (pandas, scipy, numpy, mpl, etc.)

Can you please dedent (undent? outdent?) the docstring so that it's the same indentation as the `"""` either like this?
```python
    """Test get_psm3() for multiple error scenarios:
    * Bad api key -> HTTP 403 forbidden because api_key is rejected
    * Bad latitude/longitude -> Coordinates were not found in the NSRDB
    * Bad name -> names is not one of the available options
    * Bad interval, single year -> intervals can only be 30 or 60 minutes
    """
```
or imho even better would be this:
```python
    """
    Test get_psm3() for multiple error scenarios:
    * Bad api key -> HTTP 403 forbidden because api_key is rejected
    * Bad latitude/longitude -> Coordinates were not found in the NSRDB
    * Bad name -> names is not one of the available options
    * Bad interval, single year -> intervals can only be 30 or 60 minutes
    """
```
thanks!

Reply:
I dedented the bullets to be in line and conformed it more closely to [PEP 257](https://www.python.org/dev/peps/pep-0257/). The second suggestion doesn't conform to this PEP with the extra \n at the beginning. 

Please let me know if there's a better way to list the scenarios.

Reply:
[PEP 8 says consistency within a project is more important, and PEP 20 says readability counts.](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds). Take a look at docstrings in pvlib and the other major projects, and you'll see both of the styles I listed above are in common usage. I personally prefer the leading newline, unless the docstring is a one-liner, because I like the first line to be just as readable as the following ones, and I think the quotation marks detract from the readability. Anyway my point is, I don't think we have to follow PEP 257 verbatim.

Reply:
my 2 cents: I'm really happy to see the explanation for the test. But let's not worry too much about doc string formatting in the test suite. Or use comments and avoid any debate!